vector/src/main/java/fr/gouv/tchap/features/home/contact/list/TchapContactListController.kt

```diff
@@ -0,0 +1,236 @@
+            }
+        })
+
+        tchapContactList.forEach { item ->
+            when (item) {
+                is User -> {
+                    userDirectoryUserItem {
+                        id(item.userId)
+                        selected(false)
+                        matrixItem(item.toMatrixItem())
+                        avatarRenderer(avatarRenderer)
+                        clickListener { _ ->
+                            callback?.onItemClick(item)
+                        }
+                    }
+                }
+
+                is MappedContact -> {
+                    contactItem {
+                        id(item.id)
+                        mappedContact(item)
+                        avatarRenderer(avatarRenderer)
+                    }
+                    item.emails
+                            .forEachIndexed { index, it ->
+                                if (it.matrixId == null) return@forEachIndexed
+                                contactDetailItem {
+                                    id("${item.id}-e-$index-${it.email}")
+                                    threePid(it.email)
+                                    matrixId(it.matrixId)
+                                    clickListener {
+                                        callback?.onMatrixIdClick(it.matrixId)
+                                    }
+                                }
+                            }
+                }
+            }
+        }
+    }
+
+//    private fun buildKnownUsers(currentState: TchapContactListViewState, selectedUsers: List<String>) {
+//        currentState.knownUsers()
+//                ?.filter { it.userId != session.myUserId }
+//                ?.let { userList ->
+//                    userListHeaderItem {
+//                        id("known_header")
+//                        header(stringProvider.getString(R.string.direct_room_user_list_known_title))
+//                    }
+//
+//                    if (userList.isEmpty()) {
+//                        renderEmptyState()
+//                        return
+//                    }
+//                    userList.forEach { item ->
+//                        val isSelected = selectedUsers.contains(item.userId)
+//                        userDirectoryUserItem {
+//                            id(item.userId)
+//                            selected(isSelected)
+//                            matrixItem(item.toMatrixItem())
+//                            avatarRenderer(avatarRenderer)
+//                            clickListener { _ ->
+//                                callback?.onItemClick(item)
+//                            }
+//                        }
+//                    }
+//                }
+//    }
+
+    private fun buildDirectoryUsers(directoryUsers: List<User>, searchTerms: String, ignoreIds: List<String>) {
+        val toDisplay = directoryUsers
+                .filter { !ignoreIds.contains(it.userId) && it.userId != session.myUserId }
```

Comment:
What do you think about: `filterNot { ignoreIds.contains(it.userId) || it.userId == session.myUserId }` ?

Reply:
Yes I prefer your proposal. Thanks